OpenIPSL/Electrical/Sources/SourceBehindImpedance/BaseClasses/baseVoltageSource.mo

```diff
@@ -0,0 +1,126 @@
+  parameter OpenIPSL.Types.PerUnit R_a=1e-3 "Internal source resistance, pu, system base"
+    annotation (Dialog(group="Voltage Source parameters"));
+  parameter OpenIPSL.Types.PerUnit X_d=0.2 "Internal source d-axis reactance, pu, system base"
+    annotation (Dialog(group="Voltage Source parameters"));
+
+ // Terminal Voltage variables
+  OpenIPSL.Types.PerUnit V(start=v_0) "Bus voltage magnitude";
+  OpenIPSL.Types.Angle anglev(start=angle_0) "Bus voltage angle";
+
+  // Public Auxiliary variables
+  OpenIPSL.Types.PerUnit P(start=P_0/S_b) "Active power (system base)";
+  OpenIPSL.Types.PerUnit Q(start=Q_0/S_b) "Reactive power (system base)";
+
+  // Public Internal voltage source variables (to be used as outputs for intialization of linked components, e.g. droop controller)
+  OpenIPSL.Types.Angle delta(start=delta0) "Internal voltage source angle";
+  OpenIPSL.Types.PerUnit E(start=E0) "Internal voltage source magnitude";
+
+protected
+
+  // Change of base
+  parameter Real CoB=M_b/S_b "Change from system to machine base";
+  parameter Real p0=P_0/M_b
+    "Initial active power (machine base)";
+  parameter Real q0=Q_0/M_b
+    "Initial reactive power (machine base)";
+
+  // Initialization values
+  parameter OpenIPSL.Types.PerUnit vr0=v_0*cos(angle_0);
+  parameter OpenIPSL.Types.PerUnit vi0=v_0*sin(angle_0);
+  parameter OpenIPSL.Types.PerUnit ir0=-CoB*(p0*vr0 + q0*vi0)/(vr0^2 + vi0^2);
+  parameter OpenIPSL.Types.PerUnit ii0=-CoB*(p0*vi0 - q0*vr0)/(vr0^2 + vi0^2);
+
+  // Initialization of voltage source parameters
+  parameter OpenIPSL.Types.PerUnit Er0 = vr0 + CoB*R_a*ir0 - CoB*X_d*ii0 "Initial value of the real part of the internal voltage source phasor";
+  parameter OpenIPSL.Types.PerUnit Ei0 = vi0 + CoB*R_a*ii0 + CoB*X_d*ir0 "Initial value of the imaginary part of the internal voltage source phasor";
+  parameter OpenIPSL.Types.PerUnit E0  = sqrt(Er0^2+Ei0^2) "Initial value of the internal voltage source phasor magnitude";
+  parameter OpenIPSL.Types.Angle delta0 = atan2(Ei0, Er0) "Initial value of the internal voltage source phasor angle";
+
+  // Internal voltage source variables
+  Real Er(start=Er0) "Internal voltage source, real part";
+  Real Ei(start=Ei0) "Internal voltage source, imaginary part";
+equation
+  // Equations linking the internal voltage source to the terminal bus and the pin
+  Er = p.vr + CoB*R_a*(-p.ir) - CoB*X_d*(-p.ii);
+  Ei = p.vi + CoB*R_a*(-p.ii) + CoB*X_d*(-p.ir);
+  //Power injections
+  -P = p.vr*p.ir + p.vi*p.ii;
+  -Q = p.vi*p.ir - p.vr*p.ii;
+  // Terminal voltage magnitude and angle
+  V = sqrt(p.vr^2 + p.vi^2);
+  anglev = atan2(p.vi, p.vr);
+  // Assining variables to outputs
+  Emag = sqrt(Er^2+Ei^2);
+  Edelta = atan2(Ei,Er);
+  Emag0 = E0;
+  Eang0 = delta0;
+  annotation (Icon(graphics={
+        Ellipse(extent={{-100,100},{100,-100}},
+                                           lineColor={0,0,255},
+          lineThickness=1)}),
+           preferredView="text",
```

Comment:
Sure you want to default to the text (i.e., Modelica source code) view here?

Reply:
In this case yes, because the diagram layer doesn't contain any of the implementation, so it is better for me as a user to be able to be directed to where the actual implementation is done.